data_collection_scripts/big_query_balance_data.py

```diff
@@ -72,24 +76,63 @@ def collect_data(ledgers, snapshot_dates, force_query):
                     writer.writerow([field.name for field in rows.schema])
                     writer.writerows(rows)
                 logging.info(f'Done writing {ledger} data to file.\n')
+                ledger_last_updates[ledger] = date
+    return ledger_last_updates
+
+
+def get_from_dates(granularity):
+    """
+    Get the dates from which to start querying for each ledger, which corresponds to the last updated date + the granularity
+    (e.g. the month following the last update if granularity is 'month').
+    :param granularity: The granularity of the data collection. Can be 'day', 'week', 'month', or 'year'.
+    :return: A dictionary with ledgers as keys and the corresponding start dates as values.
+    """
+    with open(hlp.ROOT_DIR / "data_collection_scripts/last_update.json") as f:

```

Comment:
If the file doesn't exist, this will throw an error. If that's intended OK, otherwise I suggest you create a separate function which checks if the files exists and, if not, initializes it somehow (eg. reading the files in the input folder).